### ISSUE TITLE
Backport better error messages?

### ISSUE THREAD

**Issue by joshuar** [logstash-plugins/logstash-filter-date#30]
Filing this on behalf of a customer.  In Logstash 1.4.x, this filter produces confusing error messages when the timestamp doesn't match the configured patterns:

```
 {:timestamp=>"2015-08-03T06:50:24.078000-0700", :message=>"Failed parsing date from field", :field=>"timestamp", :value=>"03 Aug 06:50:23.913", :exception=>java.lang.IllegalArgumentException: Invalid format: "03 Aug 06:50:23.913" is malformed at " Aug 06:50:23.913", :level=>:warn}
```

Would be nice if the raw exception was reinterpreted as is done in Logstash 1.5.x:

```
Failed parsing date from field {:field=>"timestamp", :value=>"03 Aug 06:50:23.913", :exception=>"Invalid format: \"03 Aug 06:50:23.913\"", :config_parsers=>"MMM dd YYYY HH:mm:ss.SSS", :config_locale=>"default=en_AU", :level=>:warn}
```

This makes it much easier for customers to understand what has gone wrong as it clearly indicates, the field, the value and the pattern in question that are causing issues.

**Comment by suyograo** [comment]
We will not be backporting any fixes to 1.4 -- only the critical security issues. Any reason why the user can't upgrade to 1.5?

**Comment by purbon** [comment]
Closing this as it looks stale since August. happy to reopen if something change about backporting.
